lib/IDE/Formatting.cpp

```diff
@@ -285,17 +307,16 @@ class FormatContext {
       return false;
 
     if (TInfo) {
-      if (TInfo.StartOfLineTarget->getKind() == tok::l_brace &&
-          isKeywordPossibleDeclStart(*TInfo.StartOfLineBeforeTarget) &&
-          TInfo.StartOfLineBeforeTarget->isKeyword())
+      if (TInfo.getLineStarter()->getKind() == tok::l_brace &&
+          isKeywordPossibleDeclStart(*TInfo.getLineStarter(1)) &&
```

Comment:
Should there be a check to make sure that there is actually a line with index `1`? If the method returned null you could check for null.